rust/src/lib.rs

```diff
@@ -48,20 +51,8 @@ impl GraphWrapper {
         }
     }
 
-    pub fn get_modules(&self) -> HashSet<String> {
-        self._graph
-            .all_modules()
-            .visible()
-            .names_as_strings()
-            .collect()
-    }
-
-    pub fn contains_module(&self, name: &str) -> bool {
-        match self.get_visible_module_by_name(name) {
-            Ok(_) => true,
-            Err(GrimpError::ModuleNotPresent(_)) => false,
-            _ => panic!("unexpected error checking for module existence"),
-        }
+    pub fn get_modules(&self) -> GraphModuleNames {
+        self._graph.all_modules().visible().names().collect()
```

Comment:
There's a clone/copy going on in here of all the module names (as integer symbols) - this still tanks the performance (less bad, but still bad).